http-service/chat/chat-handler.go

```diff
@@ -1,28 +1,93 @@
 package chat
 
 import (
-	"io"
 	"log"
 
+	"github.com/SteveRusin/go_mentoring/http-service/messages"
 	"golang.org/x/net/websocket"
 )
 
-type chatHandlers struct{}
+type chatHandlers struct {
+	connectionsByUser  map[string]*websocket.Conn
+	messagesRepository *messages.MessageRepository
+}
 
 func NewChatHandlers() *chatHandlers {
-	return &chatHandlers{}
+	return &chatHandlers{
+		connectionsByUser:  make(map[string]*websocket.Conn),
+		messagesRepository: messages.NewMessageRepository(),
+	}
 }
 
 func (h *chatHandlers) Connect(ws *websocket.Conn) {
+	// ask how this casting works
+	user := ws.Request().Context().Value("user").(string)
```

Comment:
it can panic here if a type is not string or the key is not present at all
https://go.dev/play/p/drIry1y5mMx